decimal.go

```diff
@@ -469,13 +469,30 @@ func (d Decimal) MarshalJSON() ([]byte, error) {
 
 // Scan implements the sql.Scanner interface for database deserialization.
 func (d *Decimal) Scan(value interface{}) error {
-	str, err := unquoteIfQuoted(value)
-	if err != nil {
+	var err error
+	// first try to see if the data is stored in database as a Numeric datatype
+	switch value.(type) {
+
+	case float64:
+		// numeric in sqlite3 sends us float64
+		*d = NewFromFloat(value.(float64))
+		return err
```

Comment:
you should return nil here, instead of the empty err above. Returning the err makes me have to find the definition of the err and make sure that it was never set to anything in order to parse this code.